driver-core/src/main/com/mongodb/internal/operation/ClientBulkWriteOperation.java

```diff
@@ -309,34 +314,30 @@ private List<List<BsonDocument>> exhaustBulkWriteCommandOkResponseCursor(
             final Connection connection,
             final BsonDocument response) {
         int serverDefaultCursorBatchSize = 0;
-        try (BatchCursor<BsonDocument> cursor = cursorDocumentToBatchCursor(
+        try (CommandBatchCursor<BsonDocument> cursor = cursorDocumentToBatchCursor(
                 TimeoutMode.CURSOR_LIFETIME,
                 response,
                 serverDefaultCursorBatchSize,
                 codecRegistry.get(BsonDocument.class),
                 options.getComment().orElse(null),
                 connectionSource,
                 connection)) {
+            cursor.setCloseWithoutTimeoutReset(true);
```

Comment:
`BULK-TODO` Why is this needed?

P.S. If undoing this, roll `SyncOperationHelper.cursorDocumentToBatchCursor` back.

Reply:
Asked a relevant question here https://mongodb.slack.com/archives/C035ZJL6CQN/p1726019394102369.

Reply:
`BULK-TODO` We need to implement https://jira.mongodb.org/browse/JAVA-5615 in `master` for this PR.

Reply:
Moved to https://github.com/mongodb/mongo-java-driver/pull/1509/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#r1861036281.